### PR TITLE
[TACHYON-1255] better hdfsUnderFs list

### PR DESCRIPTION
**Pull request by siyangy**
https://tachyon.atlassian.net/browse/TACHYON-1255

### CODE REVIEW

**Comment by calvinjia** [comment]
@siyangy Thanks for the fix, could you write a unit test verifying this change?

**Comment by siyangy** [comment]
Did you mean adding a specific test for Format? I think the basic list functionality is already covered.

**Comment by calvinjia** [comment]
@siyangy I was thinking of a test verifying the `list` behavior

**Comment by siyangy** [comment]
Is it covered by UnderStorageSystemInterfaceIntegrationTest already?

**Comment by calvinjia** [comment]
@siyangy ideally we would have something that failed with the original code and now passes with the updated code. Would this be too difficult to implement?

**Comment by siyangy** [comment]
I would think so, I had some comments on https://tachyon.atlassian.net/browse/TACHYON-1255.

**Comment by calvinjia** [comment]
@siyangy If its takes too much effort to justify blocking this PR, could you open a ticket to track the issue? I think the fix is good.

LGTM

File: underfs/hdfs/src/main/java/tachyon/underfs/hdfs/HdfsUnderFileSystem.java [hunk 1]
```diff
@@ -294,10 +292,8 @@ public boolean isFile(String path) throws IOException {
       String[] rtn = new String[files.length];
       int i = 0;
       for (FileStatus status : files) {
-        TachyonURI filePathURI = new TachyonURI(status.getPath().toUri().toString());
-        String filePath =  NetworkAddressUtils.replaceHostName(filePathURI).toString();
         // only return the relative path, to keep consistent with java.io.File.list()
-        rtn[i ++] = filePath.substring(path.length()); // mUfsPrefix
+        rtn[i ++] =  status.getPath().getName();
```
Comment (apc999) [inline underfs/hdfs/src/main/java/tachyon/underfs/hdfs/HdfsUnderFileSystem.java]: Are we still returning the relative path after this change?

**Comment by apc999** [comment]
@siyangy I left a question regarding this change.

File: underfs/hdfs/src/main/java/tachyon/underfs/hdfs/HdfsUnderFileSystem.java (hunk 1 above)
Comment (siyangy) [inline underfs/hdfs/src/main/java/tachyon/underfs/hdfs/HdfsUnderFileSystem.java]: It still returns relative path, and I think it's safer than the substring approach which is pretty fragile and can lead to the issue mentioned on the ticket.

**Comment by siyangy** [comment]
@calvinjia, I've created https://tachyon.atlassian.net/browse/TACHYON-1347 to track the test.

**Comment by calvinjia** [comment]
LGTM, thanks for the fix.

**Pull request merged**